client/src/components/content/CharacterShow.js

```diff
@@ -1,37 +1,75 @@
 import React, { useState, useEffect } from "react"
+import { useParams } from "react-router-dom";
+import VoteButtons from "./VoteButtons.js"
 
 const CharacterShow = (props) => {
   const [character, setCharacter] = useState({})
-  
+  const [totalVotes, setTotalVotes] = useState(0)
+  const [userVote, setUserVote] = useState(false)
+  const [clicked, setClicked] = useState(0)
+  const params = useParams()
+
   const getCharacter = async () => {
-    const characterId = props.match.params.id
+    const characterId = params.id
     try {
       const response = await fetch(`/api/v1/characters/${characterId}`)
       if (!response.ok) {
         const errorMessage = `${response.status} (${response.statusText})`
         const error = new Error(errorMessage)
-        throw(error)
+        throw (error)
       }
       const body = await response.json()
       setCharacter(body.character)
-    }catch(error) {
+      setTotalVotes(body.character.votes[0].sum)
+      setUserVote(body.character.voted)
+    } catch (error) {
       console.error("error in character show", error.Message)
     }
   }
-  
-  useEffect (()=> {
+
+ 
+
+  useEffect(() => {
     getCharacter()
-  }, [])
+  }, [clicked])
+
+  const addVote = async (event) => {
+    event.persist();
+    const newVote = {
+      voteValue: parseInt(event.currentTarget.value)
+    }
+    const characterId = params.id
+    try {
+      await fetch(`/api/v1/characters/${characterId}/vote`, {
+        method: 'POST',
+        headers: new Headers({
+          'Content-Type': 'application/json'
+        }),
+        body: JSON.stringify(newVote)
+      })
+      setClicked((clicked + 1))
```

Comment:
We should be setting the new vote value here when we get the response back, rather than right now it looks like upon a successful addVote, the useEffect is going to fetch the data again, which we want to avoid. We already made a request to the router for info, we can set it up to respond back with the update data that we need, rather than issuing an additional fetch request after this post

Reply:
We are not persisting the value of the user's vote, but returning the sum of the character's total votes, which would require returning to the database to re-sum the updated totals.

Reply:
This clicked state is to cause the page to re-render, thus showing the updated display of votes as well as coloring the up/down arrows with the user's submission.